src/main/java/randoop/generation/GrtNumericFuzzer.java

```diff
@@ -0,0 +1,129 @@
+import java.util.HashMap;
+import java.util.Map;
+import randoop.main.GenInputsAbstract;
+import randoop.main.RandoopBug;
+import randoop.operation.PlusOperation;
+import randoop.operation.TypedTermOperation;
+import randoop.sequence.Sequence;
+import randoop.sequence.Statement;
+import randoop.types.PrimitiveType;
+import randoop.types.PrimitiveTypes;
+import randoop.types.Type;
+import randoop.types.TypeTuple;
+import randoop.util.Randomness;
+import randoop.util.SimpleArrayList;
+
+/**
+ * Fuzzer for primitive numeric values and {@code char}. It duplicates the last value, samples a
+ * Gaussian delta <em>g ~ N(0,sigma^2)</em> (sigma configurable via {@link
+ * GenInputsAbstract#grt_fuzzing_stddev}), and appends a shared {@code +} operation to obtain <code>
+ * value + g</code>.
+ */
+@SuppressWarnings("NotJavadoc")
+public final class GrtNumericFuzzer extends GrtBaseFuzzer {
+
+  /* --------------------------- Singleton --------------------------- */
+  /** Singleton instance. */
+  private static final GrtNumericFuzzer INSTANCE = new GrtNumericFuzzer();
+
+  /**
+   * Obtain the singleton instance of {@link GrtNumericFuzzer}.
+   *
+   * @return the singleton instance
+   */
+  public static GrtNumericFuzzer getInstance() {
+    return INSTANCE;
+  }
+
+  /** Private constructor to enforce singleton. */
+  private GrtNumericFuzzer() {
+    /* no-op */
+  }
+
+  /* ------------------------------- Constants ------------------------------ */
+  /** Standard deviation sigma for Gaussian fuzzing of numeric types. */
+  private static final double GAUSSIAN_STD = GenInputsAbstract.grt_fuzzing_stddev;
+
+  /** Shared {@link PlusOperation} instance (stateless). */
+  private static final PlusOperation PLUS_OP = new PlusOperation();
+
+  /** Cache: for each primitive numeric type, a prepared <code>x+y</code> statement. */
+  private static final Map<PrimitiveType, Statement> PLUS_STMTS;
+
+  static {
+    Map<PrimitiveType, Statement> m = new HashMap<>();
+    for (Class<?> c :
+        Arrays.asList(
+            byte.class,
+            short.class,
+            char.class,
+            int.class,
+            long.class,
+            float.class,
+            double.class)) {
+      PrimitiveType pt = PrimitiveType.forClass(c);
+      m.put(pt, createPlusStatement(pt));
+    }
+    PLUS_STMTS = Collections.unmodifiableMap(m);
+  }
+
+  /* ------------------------------- API ------------------------------------ */
+  @Override
+  public boolean canFuzz(Type type) {
+    Class<?> runtimeClass = type.getRuntimeClass();
+    return runtimeClass != void.class
+        && runtimeClass != boolean.class
+        && runtimeClass != Boolean.class
+        && (runtimeClass.isPrimitive() || PrimitiveTypes.isBoxedPrimitive(runtimeClass));
+  }
+
+  @Override
+  public Sequence fuzz(Sequence sequence) {
+    Type inputType = sequence.getLastVariable().getType();
+    try {
+      // 1. Re-emit last value as a literal to avoid repeating impure stmt
```

Comment:
I don't understand why it needs to be re-emitted.  Isn't the value in the last variable of the sequence, and can't that variable be used?